local_chainalysis.go

```diff
@@ -0,0 +1,131 @@
+package chainalysis
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"io"
+	"os"
+
+	"github.com/0xsequence/ethkit/ethcoder"
+	"github.com/0xsequence/ethkit/ethrpc"
+	"github.com/0xsequence/ethkit/go-ethereum/common"
+	"github.com/0xsequence/go-sequence/lib/prototyp"
+)
+
+//go:embed data/sanctioned_addresses.json
+var sanctionedAddressesFile []byte
+
+type sanctionedAddressEvent struct {
+	BlockNum  uint64          `json:"blockNum"`
+	BlockHash string          `json:"blockHash"`
+	Addrs     []prototyp.Hash `json:"addrs"`
+}
+
+type localChainAnlysis struct {
+	SanctionedAddresses map[string]struct{}
+}
+
+func NewLocalChainAlysis() (ChainAlysis, error) {
+	var sanctionedAddresses map[string]struct{} = nil
+	sanctionedAddressEvents := []sanctionedAddressEvent{}
+
+	err := json.Unmarshal(sanctionedAddressesFile, &sanctionedAddressEvents)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, event := range sanctionedAddressEvents {
+		for _, addr := range event.Addrs {
+			sanctionedAddresses[addr.String()] = struct{}{}
+		}
+	}
+
+	return &localChainAnlysis{
+		SanctionedAddresses: sanctionedAddresses,
+	}, nil
+}
+
+func (l *localChainAnlysis) IsSanctioned(address string) (bool, error) {
+	formattedAddress := common.HexToAddress(address).Hex()
+	_, ok := l.SanctionedAddresses[formattedAddress]
+	return ok, nil
+}
+
+func FetchAndUpdateSanctionedAddresses(file *os.File, startingBlock uint64, endingBlock uint64) error {
+	fileData, err := io.ReadAll(file)
```

Comment:
instead of storing locally as it'll get outdated quickly.. put it in this repo on github, under folder "index/sanctioned_addresses.json" and add a Makefile to update this list. Then we can commit it.

Additionally, in this library, fetch it from github raw file, this way, we have a fixed set which can be updated, but then when the lib spins up, it'll locally get the latest info, etc.